pkg/component/spec.go

```diff
@@ -81,10 +81,24 @@ func (t *CommandTimeoutSpec) InitDefaults() {
 	t.Stop = 30 * time.Second
 }
 
+// ServiceTimeoutSpec is the timeout specification for subprocess.
+type ServiceTimeoutSpec struct {
+	Checkin time.Duration `config:"checkin" yaml:"checkin"`
+	Stop    time.Duration `config:"stop" yaml:"stop"`
+}
+
+// InitDefaults initialized the defaults for the timeouts.
+func (t *ServiceTimeoutSpec) InitDefaults() {
+	t.Checkin = 30 * time.Second
+	t.Stop = 3 * time.Minute
+}
+
 // ServiceSpec is the specification for an input that executes as a service.
 type ServiceSpec struct {
+	CPort      int                   `config:"cport" yaml:"cport" validate:"required"`
```

Comment:
What is CPort? Control protocol port?

Reply:
```creds port``` or ```connection info port```, didn't want it to be confused with any other possible primary connection parameters between the agent and the endpoint